### PR TITLE
Allow numbers to be used as values in styleMap()

### LINKED ISSUE #3787: [lit-html] styleMap breaks if value is a number
### Which package(s) are affected?

Lit Core (lit / lit-html / lit-element / reactive-element)

### Description

While adding a check for !important in styleMap (line 104) in version 2.7.1 it is not checked if this value is a **string**. In case of z-index the value will be a number and the check will crash as 'endsWith' naturally expects a string.

In our system we're using an older fixed version of Lit (2.6.x) but lit always grabs the latest version for lit-element or lit-html due to ^ which is not what you would expect. Even while setting a fixed version npm i crashed our applications as it installed the latest versions as dependency. Would be better to fix the version for lit sub-packages to be sure to get always the version you request.

Thanks :)

### Reproduction

Use z-index with a number.

### Workaround

Adding lit-html in version 2.7.0 or lower as dependency

### Is this a regression?

Yes. This used to work, but now it doesn't.

### Affected versions

2.7.1

### Browser/OS/Node environment

not relevant

### PR DESCRIPTION
**Pull request by justinfagnani**
Fixes #3787

### CODE REVIEW

File: .changeset/slimy-clouds-roll.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'lit-html': patch
+'lit': patch
+'lit-element': patch
```
Comment (augustjk) [inline .changeset/slimy-clouds-roll.md:4]: This isn't touching anything in `lit-element`. I think it's okay not to bump that but keep `lit`?
```suggestion
'lit-html': patch
'lit': patch
```

File: packages/lit-html/src/test/directives/style-map_test.ts [hunk 2]
```diff
@@ -79,12 +79,22 @@ suite('styleMap', () => {
...
-    renderStyleMap({marginTop: '4px', paddingBottom: '8px', opacity: '0.55'});
+    renderStyleMap({
+      marginTop: '4px',
+      paddingBottom: '8px',
+      opacity: '0.55',
+      'z-index': 1,
+    });
     assert.equal(el.style.marginTop, '4px');
     assert.equal(el.style.paddingBottom, '8px');
     assert.equal(el.style.opacity, '0.55');
```
Comment (augustjk) [inline packages/lit-html/src/test/directives/style-map_test.ts:100]:
```suggestion
    assert.equal(el.style.opacity, '0.55');
    assert.equal(el.style.zIndex, '1');
```

**Review by augustjk — CHANGES_REQUESTED** [review]

File: .changeset/slimy-clouds-roll.md (hunk 1 above)
Comment (justinfagnani) [reply .changeset/slimy-clouds-roll.md:4]: I think we need to revisit our convention here and write down the decision and rationale, so it's easy to recall.

File: packages/tests/src/web-test-runner.config.ts [hunk 3]
```diff
@@ -24,8 +24,8 @@ const browserPresets = {
   // Default set of Playwright browsers to test when running locally.
   local: [
     'chromium', // keep browsers on separate lines
-    'firefox', // to make it easier to comment out
-    'webkit', // individual browsers
+    // 'firefox', // to make it easier to comment out
+    // 'webkit', // individual browsers
```
[the remark this answers is not in the document]
Comment (justinfagnani) [reply packages/tests/src/web-test-runner.config.ts:28]: Oops!

File: packages/lit-html/src/test/directives/style-map_test.ts [hunk 4]
```diff
@@ -79,12 +79,22 @@ suite('styleMap', () => {
...
+    renderStyleMap({
+      marginTop: '2px',
+      'padding-bottom': '4px',
+      opacity: '0.5',
+      'z-index': 10,
+    });
     const el = container.firstElementChild as HTMLElement;
     assert.equal(el.style.marginTop, '2px');
     assert.equal(el.style.paddingBottom, '4px');
     assert.equal(el.style.opacity, '0.5');
```
[the remark this answers is not in the document]
Comment (justinfagnani) [reply packages/lit-html/src/test/directives/style-map_test.ts:91]: Maybe... the fact that it type-checks and doesn't throw is already a big step up in coverage.

**Review by augustjk — APPROVED** [review]

**Pull request merged**